tools/kagami/src/swarm.rs

```diff
@@ -464,6 +475,12 @@ impl AsRef<OsStr> for AbsolutePath {
     }
 }
 
+impl From<PathBuf> for AbsolutePath {
+    fn from(path: PathBuf) -> Self {
+        Self { path }
+    }
+}
```

Comment:
I used `AbsolutePath` newtype following "parse-not-validate" pattern. In other words, my intention was that `AbsolutePath` should be constructed only with `PathBuf` which is absolute. Here `AbsolutePath` is constructed without any validation.

I would suggest changing it to `TryFrom` and put resolution logic inside. It is essentially what `AbsolutePath::absolutize` is.

Reply:
If it weren't for the Rust bureaucracy, I'd recommend using the standard library methods.